engine/classes/Elgg/BootService.php

```diff
@@ -0,0 +1,227 @@
+		foreach (['__site_secret__', 'default_site', 'dataroot'] as $key) {
+			$datalist_cache->put($key, $datalists[$key]);
+		}
+		_elgg_services()->datalist->setCache($datalist_cache);
+
+		$CONFIG->site = $data->getSite();
+		$CONFIG->wwwroot = $CONFIG->site->url;
+		$CONFIG->sitename = $CONFIG->site->name;
+		$CONFIG->sitedescription = $CONFIG->site->description;
+		$CONFIG->url = $CONFIG->wwwroot;
+
+		_elgg_services()->subtypeTable->setCachedValues($data->getSubtypeData());
+
+		foreach ($data->getConfigValues() as $key => $value) {
+			$CONFIG->$key = $value;
+		}
+
+		_elgg_services()->plugins->setBootPlugins($data->getActivePlugins());
+
+		_elgg_services()->pluginSettingsCache->setCachedValues($data->getPluginSettings());
+
+		if (!$GLOBALS['_ELGG']->simplecache_enabled_in_settings) {
+			$simplecache_enabled = $datalist_cache->get('simplecache_enabled');
+			$CONFIG->simplecache_enabled = ($simplecache_enabled === false) ? 1 : $simplecache_enabled;
+		}
+
+		$system_cache_enabled = $datalist_cache->get('system_cache_enabled');
+		$CONFIG->system_cache_enabled = ($system_cache_enabled === false) ? 1 : $system_cache_enabled;
+
+		// needs to be set before system, init for links in html head
+		$CONFIG->lastcache = (int)$datalist_cache->get("simplecache_lastupdate");
+
+		$GLOBALS['_ELGG']->i18n_loaded_from_cache = false;
+
+		if (!empty($CONFIG->debug)) {
+			_elgg_services()->logger->setLevel($CONFIG->debug);
+			_elgg_services()->logger->setDisplay(true);
+		}
+
+		$GLOBALS['_ELGG']->view_path = \Elgg\Application::elggDir()->getPath("/views/");
+
+		// finish boot sequence
+		_elgg_session_boot();
+		_elgg_services()->systemCache->loadAll();
+		_elgg_services()->translator->loadTranslations();
+
+		// we always need site->email and user->icontime, so load them together
+		$preload_md_guids = [$CONFIG->site_guid];
+		$user_guid = _elgg_services()->session->getLoggedInUserGuid();
+		if ($user_guid) {
+			$preload_md_guids[] = $user_guid;
+		}
+		_elgg_services()->metadataCache->populateFromEntities($preload_md_guids);
+
+		// TODO get rid of this, then we can drop the metadata preload
+		$CONFIG->siteemail = $CONFIG->site->email;
+
+		// gives hint to get() how to approach missing values
+		$CONFIG->site_config_loaded = true;
+
+		// invalidate on some actions just in case other invalidation triggers miss something
+		_elgg_services()->hooks->registerHandler('action', 'all', function ($action) {
+			if (0 === strpos($action, 'admin/' || $action === 'plugins/settings/save')) {
+				$this->invalidateCache();
+			}
+		}, 1);
+	}
+
+	/**
+	 * Invalidate the cache item
+	 *
+	 * @param int $site_guid Site GUID or empty for current site
+	 *
+	 * @return void
+	 */
+	public function invalidateCache($site_guid = 0) {
+		$CONFIG = _elgg_services()->config->getStorageObject();
+		if (!$site_guid) {
+			$site_guid = $CONFIG->site_guid;
+		}
+
+		// this gets called a lot on plugins page, avoid thrashing cache
```

Comment:
The plugins page always calls `\Elgg\Database\Plugins::generateEntities()` which removes the priority setting and reindexes before displaying them.